Zend/Optimizer/zend_func_info.c

```diff
@@ -892,25 +873,33 @@ ZEND_API uint32_t zend_get_func_info(
 	return ret;
 }
 
-int zend_func_info_startup(void)
+void zend_func_info_add(const func_info_t *func_infos)
 {
 	size_t i;
 
+	for (i = 0; i < sizeof(*func_infos)/sizeof(func_info_t); i++) {
```

Comment:
This doesn't do what you want it to do :) At this point `sizeof(*func_infos)` is simply `sizeof(func_info_t)`, so this will actually always only add a single function!

You should instead pass in the size as a separate parameter, where it will work on the array type, not the decayed pointer type.